wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareReaderFailoverHandler.java

```diff
@@ -372,13 +396,37 @@ private ReaderFailoverResult getNextResult(final CompletionService<ReaderFailove
     }
   }
 
-  private class ConnectionAttemptTask implements Callable<ReaderFailoverResult> {
+  protected PluginService getNewPluginService() {
+    return new PartialPluginService(
```

Comment:
What's not multi-thread safe inside PartialPluginService? I see that these two new instances of PartialPluginService shares the same ConnectionPluginManager instance. Is it ok?